hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordReader.java

```diff
@@ -241,7 +241,12 @@ private void scanInternalV1(Option<KeySpec> keySpecOpt) {
     try {
       // Iterate over the paths
       logFormatReaderWrapper = new HoodieLogFormatReader(fs,
-          logFilePaths.stream().map(logFile -> new HoodieLogFile(new CachingPath(logFile))).collect(Collectors.toList()),
+          logFilePaths.stream()
+              .map(filePath -> new HoodieLogFile(new CachingPath(filePath)))
+              // hit an uncommitted file possibly from a failed write, skip processing this one
+              .filter(logFile -> completedInstantsTimeline.containsOrBeforeTimelineStarts(logFile.getDeltaCommitTime())
```

Comment:
Should this logic be dependent on table version?

Reply:
Do we want a new table version just for the reader? The new release requires this code to eliminate the FileNotFoundExceptions.

Reply:
There are some issues I can think of:

1. if the table has been upgraded, the table version check would become invalid;
2. if there is coexistence of old and new log files, we need to filter out only the new log files, but for old log files, we do not have very good manner to check its base instant time because that might already be archived.

Reply:
Fix it by checking both the file name and the block instant. There might be some minor performace regression for the double check, but do we have better solution?